### PR TITLE
fix: Intl.NumberFormat locale RangeError related to Simulation Details

### LINKED ISSUE #24067: [Bug]: Simulation Details reports NumberFormat RangeError for some locales
@anaamolnar reported 3 inquiries of the error “RangeError: Incorrect locale information provided” when attempting a transaction.

```
RangeError: Incorrect locale information provided
  at Intl.NumberFormat (<anonymous>)
  at chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-6.js:1:286226
  at a.IndividualFiatDisplay (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-7.js:1:392298)
  at ga (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-1.js:33:266221)
  at _l (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-1.js:33:311148)
  at Ru (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-1.js:33:297011)
  at Au (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-1.js:33:296936)
  at Su (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-1.js:33:294276)
  at chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-1.js:33:252626
  at n.unstable_runWithPriority (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-2.js:1:335481)
  at $i (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-1.js:33:252335)
  at eo (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-1.js:33:252571)
  at Zi (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-1.js:33:252506)
  at bu (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-1.js:33:291042)
  at Ra (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-1.js:33:269878)
  at chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/ui-6.js:1:279473
```

### PR DESCRIPTION
**Pull request by dbrans**
## **Description**
**Problem:** The locale returned by `getCurrentLocale` is not meant to be used with the [Intl API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl).

**To fix this, this PR will:**
* Rename `getCurrentLocale` => `getLocaleNotSafeForIntl` to alert developers in the future.
* Add `getIntlLocale` which converts the locale returned from `getLocaleNotSafeForIntl` into a BCP 47 Language Tag which can be used with the Intl API.
* Add tests to ensure that `getIntlLocale` works with all our locales going forward.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24068?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24067

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/ducks/locale/locale.js [hunk 1]
```diff
@@ -13,7 +14,25 @@ export default function reduceLocaleMessages(state = {}, { type, payload }) {
   }
 }
 
-export const getCurrentLocale = (state) => state.localeMessages.currentLocale;
+/**
+ * @param state
+ * @returns {string} one of the codes in file://./../../../app/_locales/index.json.
+ * These codes are not safe to use with the Intl API.
+ */
+export const getLocaleNotSafeForIntl = (state) =>
```
Comment (matthewwalsh0) [inline ui/ducks/locale/locale.js:22]: For the sake of minimising refactor and the PR diff, is this really needed if `SimulationDetails` was the first usage of `Intl.NumberFormat` with one of these locales, and there was no issues previously?

Would the above warning in the JSDoc, plus the presence of the `getIntlLocale` method be a sufficient deterrent?

File: ui/ducks/locale/locale.js [hunk 2]
```diff
@@ -13,7 +14,25 @@ export default function reduceLocaleMessages(state = {}, { type, payload }) {
   }
 }
 
-export const getCurrentLocale = (state) => state.localeMessages.currentLocale;
+/**
+ * @param state
+ * @returns {string} one of the codes in file://./../../../app/_locales/index.json.
```
Comment (matthewwalsh0) [inline ui/ducks/locale/locale.js:19]: Should we be explicit in both these method's JSDoc that it's the user's selected locale that is returned?

File: ui/ducks/locale/locale.test.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import locales from '../../../app/_locales/index.json';
+import { getIntlLocale } from './locale';
+
+const createMockStateWithLocale = (locale: string) => ({
+  localeMessages: { currentLocale: locale },
+});
+
+describe('getIntlLocale', () => {
+  it.each(locales)('can convert locale "%s" to BCP 47 format', (locale) => {
```
Comment (matthewwalsh0) [inline ui/ducks/locale/locale.test.ts:9]: Love this test! But if this is the happy path, should we also have a unit test to verify we throw if a bad locale is in the state?

File: ui/ducks/locale/locale.js (hunk 1 above)
Comment (dbrans) [reply ui/ducks/locale/locale.js]: Done.

File: ui/ducks/locale/locale.js (hunk 2 above)
Comment (dbrans) [reply ui/ducks/locale/locale.js]: Done.

File: ui/ducks/locale/locale.test.ts (hunk 3 above)
Comment (dbrans) [reply ui/ducks/locale/locale.test.ts]: Good call. Done.

**Review by matthewwalsh0 — APPROVED** [review]

**Review by vinistevam — APPROVED** [review]

**Pull request merged**